src/ConstantProduct.sol

```diff
@@ -233,4 +231,32 @@ contract ConstantProduct is IConditionalOrderGenerator {
     function supportsInterface(bytes4 interfaceId) external view virtual override returns (bool) {
         return interfaceId == type(IConditionalOrderGenerator).interfaceId || interfaceId == type(IERC165).interfaceId;
     }
+
+    /**
+     * @dev Computes the difference between the two input values. If it detects
+     * an underflow, the function reverts with a custom error that informs the
+     * watchtower to poll next.
+     * If the function reverted with a standard underflow, the watchtower would
+     * stop polling the order.
+     * @param lhs The minuend of the subtraction
+     * @param rhs The subtrahend of the subtraction
+     * @return The difference of the two input values
+     */
+    function sub(uint256 lhs, uint256 rhs) internal view returns (uint256) {
+        if (lhs < rhs) {
+            revertPollAtNextBucket("subtraction underflow");
```

Comment:
nit: given that we're in control of _a_ the watch-tower (and can allocate resources), given that the price may move on chain before the end of the current bucket, and therefore affect the order returned by `getTradeableOrder`, perhaps in this case we actually just want to inform the watch-tower to poll at the next block (instead of delaying to the end of the epoch).

This may be more desirable, however the current logic (and the likelihood it comes about) is nonetheless still sound.

Reply:
My idea is that we'd follow the same error logic that we use if the amount is too low. In the end, you can only avoid a subtraction overflow (and a too-low amount) if the oracle price changes. This is just to say that I'd change this for both cases at the same time, maybe in another PR to clearly separate it from this PR.
If you don't think this overburdens the watchtower if there are many AMMs running at the same time, I don't mind making the repolling more aggressive.

Reply:
Sure, understand the same logic behind both. What we can do is indicate from the smart contract _what is the most optimal_ (and fits within the confines of the default watch-tower implementation, which try-every-block does).

Then, if we need to implement additional logic, we can do so in off-chain in the watch-tower to dial it back if it becomes too heavy :smiley:

Reply:
Agreed! Will be part of another PR.

Reply:
Implemented in #18, moving the discussion there.